Description: Get dpkg error details.
 If dpkg filed in hook actions, apt-pkg may not output useful error details,
 set Transaction errorDetails when not capture apt-pkg error info.
Forwarded: not-needed
Last-Update: 2023-04-26 

--- a/src/worker/workerinstallprogress.cpp
+++ b/src/worker/workerinstallprogress.cpp
@@ -133,6 +133,13 @@
     }
 
     res = (pkgPackageManager::OrderResult)WEXITSTATUS(ret);
+    
+    // PATCH: uniontech0004_add_feature_get_dpkg_errordetails.patch
+    // If dpkg failed in hook actions, apt-pkg may not output useful error details, 
+    // set Transaction errorDetails when not capture apt-pkg error info.
+    if (pkgPackageManager::Completed != res && m_trans->errorDetails().isEmpty()) {
+        m_trans->setErrorDetails(QString::fromUtf8(masterbuf));
+    }
 
     close(readFromChildFD[0]);
     close(readFromChildFD[1]);
